From b91a6201307b72beb522300366aad763d19b1456 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Thu, 20 Mar 2025 14:51:18 +0100
Subject: [PATCH] set upper bound in cmake_minimum_required

setting an upper bound improves forward-compatibility as legacy version support is dropped

- 3.5 compat is deprecated in 3.27 (2023), removed in 4.0 (2025)
- 3.10 compat is deprecated in 3.31 (2024)

Signed-off-by: Moritz Haase <Moritz.Haase@bmw.de>
Upstream-Status: Backport [b91a6201307b72beb522300366aad763d19b1456]
---
 CMakeLists.txt           | 4 ++--
 tests/CMakeLists.txt     | 2 +-
 unittests/CMakeLists.txt | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e4ce8a3d..3ab2259e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,9 +1,9 @@
 # CMake build script for ZeroMQ
 
 if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL Darwin)
-  cmake_minimum_required(VERSION 3.0.2)
+  cmake_minimum_required(VERSION 3.0.2...3.31)
 else()
-  cmake_minimum_required(VERSION 2.8.12)
+  cmake_minimum_required(VERSION 2.8.12...3.31)
 endif()
 
 project(ZeroMQ)
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index a8dfa8f2..eb59de29 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,5 +1,5 @@
 # CMake build script for ZeroMQ tests
-cmake_minimum_required(VERSION "2.8.1")
+cmake_minimum_required(VERSION 2.8.1...3.31)
 
 # On Windows: solution file will be called tests.sln
 project(tests)
diff --git a/unittests/CMakeLists.txt b/unittests/CMakeLists.txt
index 64198eeb..58ea875b 100644
--- a/unittests/CMakeLists.txt
+++ b/unittests/CMakeLists.txt
@@ -1,5 +1,5 @@
 # CMake build script for ZeroMQ unit tests
-cmake_minimum_required(VERSION "2.8.1")
+cmake_minimum_required(VERSION 2.8.1...3.31)
 
 set(unittests
     unittest_ypipe
